src/hdmf/backends/hdf5/h5tools.py

```diff
@@ -746,10 +755,19 @@ def close_linked_files(self):
         not, which prevents the linked-to file from being deleted or truncated. Use this method to close all opened,
         linked-to files.
         """
-        for obj in self.__open_links:
-            if obj:
-                obj.file.close()
-        self.__open_links = []
+        # Make sure
+        try:
+            for obj in self.__open_links:
+                if obj:
+                    obj.file.close()
+        except AttributeError:
+            # Do not do anything in case that self.__open_links does not exist. This
+            # may happen in case that an error occurs before HDF5IO has been fully
+            # setup in __init__, e.g,. if a child class (such as NWBHDF5IO) raises
+            # an error before self.__open_links has been created.
+            pass
+        finally:
```

Comment:
Is the `finally` necessary? I think it can be removed. If it is useful, perhaps `close` should also have a `finally`

Reply:
I think it is safer to have the ``finally`` as it helps ensure that any future operations that may happen on ``self.__open_links`` after calling the function are valid. I don't think we need a finally in ``close``, because we here only want to call ``close()`` on the file object but not modify the value of ``self.file``. However we should probably set ``self.__file = None`` when catching the error where it is missing. I'll add that change.

Reply:
I added the change to ``close``